core/common/src/Buffers.kt

```diff
@@ -22,7 +22,9 @@ public fun Buffer.snapshot(): ByteString {
         var curr = head
         do {
             check(curr != null) { "Current segment is null" }
-            append(curr.data, curr.pos, curr.limit)
+            for (idx in 0 until curr.size) {
```

Comment:
Is it not degrading performance?
It seems to me that writing byte-by-byte will be very slow due to the large number of checks and frequent `copyInto` calls

Reply:
Thanks, I'll reimplement it. We don't track `Buffer.snapshot` performance, so it didn't ring the bell once I plunked a naive implementation.